### ISSUE TITLE
CrossEntropyLoss2d

### ISSUE THREAD

**Issue by kshpv** [JindongJiang/RedNet#10]
Hello!
Thank you for your work!
I am a little bit confused about **CrossEntropyLoss2d** class, especially:
```
targets_m[mask] -= 1
loss_all = self.ce_loss(inputs, targets_m.long())
losses.append(torch.sum(torch.masked_select(loss_all, mask)) / torch.sum(mask.float()))
```
1. Why do you subtract 1 from targets masks?
2. Why do you use `torch.masked_select(loss_all, mask))` for calculating a loss for a single target? It seems incorrect because in that case we forget about a background.

Thank you!

**Comment by kshpv** [comment]
By the way, Why do you add **1** to all class labels in `                grid_image = make_grid(utils.color_label(torch.max(pred_scales[0][:3], 1)[1] + 1), 3, normalize=False,
                                       range=(0, 255))`?
Is it correct?

Thank you!
